### PR TITLE
[CodingStyle] Skip concat on first arg on ConsistentImplodeRector

### LINKED ISSUE #7903: Incorrect behavior of ConsistentImplodeRector
# Bug Report

I'm not sure how the rule should handle this case. If you try to refactor `implode('string a', 'string b' )` the rule will not change the order of the arguments, that is fine, the code will throw an error anyway.

But in my case I have a variable as second argument: `implode('  ' . $separator,  $anArray)`, `$anArray` is an array but due to some errors in annotations phpstan thinks that it is a string, so the rule is changing the order of the arguments and causing errors. I know that if I fix the problem with the annotations the rule work as expected.

So, I was thinking that the correct behavior for this case it's to not change the order of the arguments since it's something unexpected, but I'm not sure. 

I will fix the annotations in my code, this will not be a problem for me. But I wanted to report this to see if it's a case that needs to be handled.

| Subject        | Details              |
| :------------- | :--------------------|
| Rector version | last dev-main      |
| Installed as   | composer dependency  |

## Minimal PHP Code Causing Issue

See https://getrector.com/demo/ec054f63-0f4e-489d-a3da-fa87581814f0

```php
<?php

final class DemoFile
{
    public function run()
    {
        implode('|' . ' ', 'should not be changed?');
    }
}
```

### Responsible rules

* `ConsistentImplodeRector`

## Expected Behavior

The order of the arguments should not change.

### PR DESCRIPTION
**Pull request by nerones**
# Failing Test for ConsistentImplodeRector

Based on https://getrector.com/demo/ec054f63-0f4e-489d-a3da-fa87581814f0

Reported in https://github.com/rectorphp/rector/issues/7903
Closes https://github.com/rectorphp/rector/issues/7903

### CODE REVIEW

**Review by samsonasik — CHANGES_REQUESTED** [review]
Could you try provide a patch as well, the first arg value should check its type insted of value on this line:

https://github.com/rectorphp/rector-src/blob/31f64cc2d45491304065e2f04343900110005a19/rules/CodingStyle/Rector/FuncCall/ConsistentImplodeRector.php#L98-L100

should can be checked by something like this:

```php
$type = $this->getType($firstArgumentValue);
if ($type->isString()->yes()) {
     return null;
}
```

**Comment by nerones** [comment]
Thanks for making it so easy to contribute!

**Comment by samsonasik** [comment]
Please update PR title to something like:

```
[CodingStyle] Skip concat on first arg on ConsistentImplodeRector
```

File: rules-tests/CodingStyle/Rector/FuncCall/ConsistentImplodeRector/Fixture/string_arguments.php.inc [hunk 1]
```diff
@@ -0,0 +1,13 @@
+<?php
+
+namespace Rector\Tests\CodingStyle\Rector\FuncCall\ConsistentImplodeRector\Fixture;
+
+final class DemoFile
```
Comment (samsonasik) [inline rules-tests/CodingStyle/Rector/FuncCall/ConsistentImplodeRector/Fixture/string_arguments.php.inc]:
```suggestion
final class SkipConcatOnFirstArg
```

File: rules-tests/CodingStyle/Rector/FuncCall/ConsistentImplodeRector/Fixture/string_arguments.php.inc [hunk 2]
```diff
@@ -0,0 +1,13 @@
+<?php
```
Comment (samsonasik) [inline rules-tests/CodingStyle/Rector/FuncCall/ConsistentImplodeRector/Fixture/string_arguments.php.inc]: Please update fixture file to : `skip_concat_on_first_arg.php.inc`

**Review by samsonasik — APPROVED** [review]

**Review by samsonasik — APPROVED** [review]
Thank you @nerones

**Pull request merged**